docs/references/go/verifying-sessions.mdx

```diff
@@ -3,100 +3,153 @@ title: Veryifing a session | Clerk Go SDK
 description: Learn how to verify a session with Clerk in your Go application.
 ---
 
-# Verifying a session in Go
+# Verifying a Clerk session in Go
 
-## Protect your backend APIs
+Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples show how to verify a session and retrieve the corresponding user.
 
-Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples shows how to verify a session and retrieve the corresponding user.
+## Using middleware
 
-<InjectKeys>
+The library provides two functions that can be used for adding authentication with Clerk to HTTP handlers.
 
-```go
-package main
+Both middleware functions support header based authentication with a bearer token. The token is parsed, verified and
+its claims are extracted as [SessionClaims](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. The library
+provides a helper for accessing the claims from the context, [SessionClaimsFromContext](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext).
 
+The two middleware functions are [WithHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
+and [RequireHeaderAuthorization](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization).
+Their difference is that the `RequireHeaderAuthorization` calls `WithHeaderAuthorization` under the hood, but responds
+with HTTP 403 Forbidden if it fails to detect valid session claims.
+
+Let's see how the middleware functions can be used in action.
```

Comment:
This feels like a filler sentence. I'd say it can be removed and the context around the code blocks below stays the same.

Reply:
Removed.